src/corecel/sys/Device.cc

```diff
@@ -65,7 +67,9 @@ Device& global_device()
                 << "CUDA active device ID unexpectedly changed from "
                 << device.device_id() << " to " << cur_id;
             std::lock_guard<std::mutex> scoped_lock{device_setter_mutex()};
+            auto num_streams = device.num_streams();
             device = Device(cur_id);
+            device.create_streams(num_streams);
```

Comment:
This seems a little weird: switching device IDs shouldn't destroy/delete all existing streams. Since this block of code is here mostly as an error check, perhaps we should just assert if the device changes?

If we need to add the ability to access multiple devices, etc., we may need to look at the lower-level [CUDA driver API for context management](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__CTX.html)